components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AbstractComplexController.java

```diff
@@ -290,23 +306,25 @@ public void save(Patient patient)
             XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
             BaseObject dataHolder = doc.getXObject(getXClassReference());
             PatientData<T> data = patient.getData(this.getName());
-            if (dataHolder == null && data != null) {
+            if (dataHolder == null || dataHolder == null) {
```

Comment:
I think with that change - https://github.com/phenotips/phenotips/pull/2047/commits/9aa4f6ae0423e0765ea08eb3988df7de22b5d30f#diff-a359c7d8e774fa88f6b015042ad1f9a0R179 - data will actually be null, and the old code was correct?